joblib/numpy_pickle.py

```diff
@@ -334,44 +241,35 @@ def load_build(self):
         dispatch[pickle.BUILD] = load_build
 
 
-class ZipNumpyUnpickler(NumpyUnpickler):
-    """A subclass of our Unpickler to unpickle on the fly from
-    compressed storage."""
-
-    def __init__(self, filename, file_handle):
-        NumpyUnpickler.__init__(self, filename,
-                                file_handle,
-                                mmap_mode=None)
-
-    def _open_pickle(self, file_handle):
-        return BytesIO(read_zfile(file_handle))
-
-
 ###############################################################################
 # Utility functions
 
-def dump(value, filename, compress=0, cache_size=100, protocol=None):
-    """Fast persistence of an arbitrary Python object into one or multiple
-    files, with dedicated storage for numpy arrays.
+def dump(value, filename, compress=0, protocol=None, cache_size=None):
+    """Persist an arbitrary Python object.
+
+    Fast persistence of an arbitrary Python object into one file with
+    dedicated storage for numpy arrays.
 
     Parameters
     -----------
     value: any Python object
         The object to store to disk
-    filename: string
+    filename: str
         The name of the file in which it is to be stored
-    compress: integer for 0 to 9, optional
-        Optional compression level for the data. 0 is no compression.
+    compress: int from 0 to 9 or bool, optional
+        Optional compression level for the data. 0 or False is no compression.
         Higher means more compression, but also slower read and
         write times. Using a value of 3 is often a good compromise.
         See the notes for more details.
-    cache_size: positive number, optional
+        If compress is True, the compression level used is 3.
+    protocol: positive int
+        Pickle protocol, see pickle.dump documentation for more details.
+    cache_size: positive int, optional
         Fixes the order of magnitude (in megabytes) of the cache used
```

Comment:
And potentially remove the previous documentation for cache_size since it doesn't apply anymore.